subgraphs/tornado-cash/src/common/getters.ts

```diff
@@ -0,0 +1,414 @@
+  SECONDS_PER_HOUR,
+  RewardTokenType,
+  PROTOCOL_NAME,
+  PROTOCOL_SLUG,
+  PROTOCOL_SCHEMA_VERSION,
+  PROTOCOL_SUBGRAPH_VERSION,
+  PROTOCOL_METHODOLOGY_VERSION,
+  ETH_ADDRESS,
+  ETH_NAME,
+  ETH_SYMBOL,
+  ETH_DECIMALS,
+  TORN_ADDRESS,
+  TORN_NAME,
+  TORN_SYMBOL,
+  TORN_DECIMALS,
+  BNB_ADDRESS,
+  BNB_DECIMALS,
+  BNB_NAME,
+  BNB_SYMBOL,
+} from "./constants";
+import { getUsdPricePerToken } from "../prices";
+import { addToArrayAtIndex } from "../common/utils/arrays";
+import { bigIntToBigDecimal } from "./utils/numbers";
+
+import { TornadoCashETH } from "../../generated/TornadoCash01/TornadoCashETH";
+import { TornadoCashERC20 } from "../../generated/TornadoCash01/TornadoCashERC20";
+import { TornadoCashBNB } from "../../generated/TornadoCash01/TornadoCashBNB";
+import {
+  Protocol,
+  Pool,
+  PoolDailySnapshot,
+  PoolHourlySnapshot,
+  Token,
+  RewardToken,
+  UsageMetricsHourlySnapshot,
+  UsageMetricsDailySnapshot,
+  FinancialsDailySnapshot,
+} from "../../generated/schema";
+
+export function getOrCreateProtocol(): Protocol {
+  let network = dataSource.network().toUpperCase();
+  let protocol = Protocol.load(FACTORY_ADDRESS.get(network)!.toHexString());
+
+  if (!protocol) {
+    protocol = new Protocol(FACTORY_ADDRESS.get(network)!.toHexString());
+    protocol.name = PROTOCOL_NAME;
+    protocol.slug = PROTOCOL_SLUG;
+    protocol.schemaVersion = PROTOCOL_SCHEMA_VERSION;
+    protocol.subgraphVersion = PROTOCOL_SUBGRAPH_VERSION;
+    protocol.methodologyVersion = PROTOCOL_METHODOLOGY_VERSION;
+    protocol.network = network;
+    protocol.type = ProtocolType.GENERIC;
+    protocol.totalValueLockedUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO;
+    protocol.cumulativeUniqueUsers = 0;
+    protocol.totalPoolCount = 0;
+    protocol.pools = [];
+
+    protocol.save();
+  }
+
+  return protocol;
+}
+
+export function getOrCreateToken(
+  tokenAddress: Address,
+  blockNumber: BigInt
+): Token {
+  let token = Token.load(tokenAddress.toHexString());
+
+  if (!token) {
+    token = new Token(tokenAddress.toHexString());
+    let network = dataSource.network().toUpperCase();
+
+    if (tokenAddress == Address.fromString(ETH_ADDRESS)) {
+      token.name = ETH_NAME;
+      token.symbol = ETH_SYMBOL;
+      token.decimals = ETH_DECIMALS;
+    } else if (tokenAddress == TORN_ADDRESS.get(network)!) {
+      token.name = TORN_NAME;
+      token.symbol = TORN_SYMBOL;
+      token.decimals = TORN_DECIMALS;
+    } else if (tokenAddress == Address.fromString(BNB_ADDRESS)) {
+      token.name = BNB_NAME;
+      token.symbol = BNB_SYMBOL;
+      token.decimals = BNB_DECIMALS;
+    } else {
+      token.name = fetchTokenName(tokenAddress);
+      token.symbol = fetchTokenSymbol(tokenAddress);
+      token.decimals = fetchTokenDecimals(tokenAddress) as i32;
+    }
```

Comment:
Every time a new network is added, with its own native token, we should modify this part. You could generalize it like:

```
if (isNativeToken(tokenAddress)) {
  let conf = configurations.getNativeToken();
  token.name = conf.name;
  token.symbol = conf.symbol;
  token.decimals = conf.decimals;
} else if (isTornToken(tokenAddress) {
  let conf = configurations.getTornToken();
  token.name = conf.name;
  token.symbol = conf.symbol;
  token.decimals = conf.decimals;
} else {
  ...
}
```

`isNativeToken()` and `isTornToken()` should be based on configurations too, so they don't need to be modified with each new network addition